client/src/client/sync.rs

```diff
@@ -0,0 +1,189 @@
+        // Ensure that the uri passed in can parse.
+        let _: Uri = uri.parse()?;
+        Self::build(Host::Other(uri.to_string()))
+    }
+
+    fn build(host: Host) -> Result<Self> {
+        let inner = reqwest::Client::new();
+        Ok(Client { host, inner })
+    }
+
+    /// Constructs a new stellar client connected to the horizon test network.
+    ///
+    /// ## Examples
+    ///
+    /// ```
+    /// use stellar_client::sync::Client;
+    /// let client = Client::horizon_test().unwrap();
+    /// ```
+    pub fn horizon_test() -> Result<Self> {
+        Self::build(Host::HorizonTest)
+    }
+
+    /// Returns true if this is a test client.
+    ///
+    /// ## Examples
+    ///
+    /// ```
+    /// # use stellar_client::sync::Client;
+    /// let client = Client::horizon_test().unwrap();
+    /// assert!(!client.is_horizon());
+    /// assert!(client.is_horizon_test());
+    /// ```
+    pub fn is_horizon_test(&self) -> bool {
+        self.host == Host::HorizonTest
+    }
+
+    /// Constructs a new stellar client connected to the horizon test network.
+    ///
+    /// ## Examples
+    ///
+    /// ```
+    /// use stellar_client::sync::Client;
+    /// let client = Client::horizon().unwrap();
+    /// ```
+    pub fn horizon() -> Result<Self> {
+        Self::build(Host::HorizonProd)
+    }
+
+    /// Returns true if this is a horizon@stellar client.
+    ///
+    /// ## Examples
+    ///
+    /// ```
+    /// # use stellar_client::sync::Client;
+    /// let client = Client::horizon().unwrap();
+    /// assert!(client.is_horizon());
+    /// assert!(!client.is_horizon_test());
+    /// ```
+    pub fn is_horizon(&self) -> bool {
+        self.host == Host::HorizonProd
+    }
+
+    #[allow(dead_code)] // TODO: Used for joining to end points
+    fn uri<'a>(&'a self) -> &'a str {
+        match self.host {
+            Host::HorizonTest => HORIZON_TEST_URI,
+            Host::HorizonProd => HORIZON_URI,
+            Host::Other(ref uri) => uri,
+        }
+    }
+
+    /// Issues a request to the stellar horizon server synchronously.
+    ///
+    /// ## Examples
+    ///
+    /// ```
+    /// use stellar_client::sync::Client;
+    /// use stellar_client::endpoint::AccountDetails;
+    /// let client = Client::horizon_test().unwrap();
+    /// let endpoint =
+    ///     AccountDetails::new("GCEZWKCA5VLDNRLN3RPRJMRZOX3Z6G5CHCGSNFHEYVXM3XOJMDS674JZ");
+    /// let account = client.request(endpoint).unwrap();
+    /// assert_eq!(account.id(), "GCEZWKCA5VLDNRLN3RPRJMRZOX3Z6G5CHCGSNFHEYVXM3XOJMDS674JZ");
+    /// ```
+    pub fn request<E>(&self, endpoint: E) -> Result<E::Response>
+    where
+        E: EndPoint,
+    {
+        let request = endpoint.into_request(&self.uri())?;
+        let request = Self::http_to_reqwest(request);
+        let response = self.inner.execute(request)?;
+        if response.status().is_success() {
+            let resp: E::Response = serde_json::from_reader(response)?;
```

Comment:
How does this know to use the deserializer from the DeserializeOwned associated type?

Reply:
Because the associated type `E::Response` must implement the trait `DeserializeOwned` so it calls the `deserialize` method that's on the `Account` struct.